CRM/MembershipExtras/BAO/MembershipPeriod.php

```diff
@@ -269,30 +266,66 @@ private static function doesOverlapWithOtherActivePeriods($periodParams) {
     }
 
     if (!$periodIsActive) {
-      return FALSE;
+      return;
     }
 
     if (empty($periodNewStartDate)) {
       throw new CRM_Core_Exception("Can't calculate period overlapping without period start date!");
     }
 
+    $overlappedMembershipPeriods = self::getOverlappingPeriods($periodID, $membershipID, $periodNewStartDate, $periodNewEndDate);
+    if ($overlappedMembershipPeriods->N > 0) {
+      $overlappingStartDate = new DateTime($periodNewStartDate);
+      $overlappingEndDate = new DateTime($periodNewEndDate);
+      $conflictingPeriodStartDate = new DateTime($overlappedMembershipPeriods->start_date);
+      $conflictingPeriodEndDate = new DateTime($overlappedMembershipPeriods->end_date);
+      
+      $isEndDateOverlap = $overlappingEndDate > $conflictingPeriodStartDate && $overlappingStartDate < $conflictingPeriodStartDate;
+      if ($isEndDateOverlap) {
+        throw new CRM_Core_Exception(
+          'The new end date of this membership period overlaps with another
+          activated membership period. Please review your changes.'
+        );
+      }
+      
+      $isStartDateOverlap = $overlappingStartDate < $conflictingPeriodEndDate && $overlappingEndDate > $conflictingPeriodEndDate;
+      if ($isStartDateOverlap) {
+        throw new CRM_Core_Exception(
+          'The new start date of this membership period overlaps with
+          another activated membership period. Please review your changes.'
+        );
+      }
+      
+      throw new CRM_Core_Exception('Undetermined problem found validating period overlapping.');
```

Comment:
I think 

`The new membership period dates are overlapping with another period dates, please validate the new dates`

is more clear